Spanner/tests/System/BatchTest.php

```diff
@@ -106,6 +123,80 @@ public function testBatch()
         $snapshot->close();
     }
 
+    public function testBatchWithRestrictiveDatabaseRole()
+    {
+        // Emulator does not support FGAC
+        $this->skipEmulatorTests();
+        
+        $query = 'SELECT
+                id,
+                decade
+            FROM ' . self::$tableName . '
+            WHERE
+                decade > @earlyBound
+            AND
+                decade < @lateBound';
+
+        $parameters = [
+            'earlyBound' => 1960,
+            'lateBound' => 1980
+        ];
+
+        $resultSet = iterator_to_array(self::$database->execute($query, ['parameters' => $parameters]));
+
+        $batch = self::$client->batch(self::INSTANCE_NAME, self::$dbName, ['databaseRole' => self::$databaseRole]);
+        $string = $batch->snapshot()->serialize();
+
+        $snapshot = $batch->snapshotFromString($string);
+
+        try {
+            $partitions = $snapshot->partitionQuery($query, ['parameters' => $parameters]);
```

Comment:
Add a `$fail = true` flag. If no exception is thrown, then the test should fail.

Reply:
Or use, `expectException`

Reply:
Or, `$this->fail(..)`

Reply:
Done